### ISSUE TITLE
Bug: fails linking exactly same positions

### ISSUE THREAD

**Issue by yfukai** [yfukai/laptrack#29]
Using a sparse matrix with `lap`, the value `0` is assumed to be infinite, and thus not connected. One needs to add an offset to the matrix elements corresponding to finite values.

**Comment by yfukai** [comment]
This seems to be caused by `scipy` bug:
https://github.com/scipy/scipy/issues/9417
https://stackoverflow.com/questions/53072911/how-do-you-store-explicit-0-values-in-a-scipy-sparse-lil-matrix

**Comment by grfrederic** [comment]
looks like this issue reappeared at some point, just ran into this problem while using the current pip release

**Comment by yfukai** [comment]
Thanks @grfrederic for the report!

Could you provide a minimum reproducible example?

It seems this automatic test is working.
https://github.com/yfukai/laptrack/blob/a36cc10b227cede837aef82526a3e7955c60a14f/tests/test_tracking.py#L190-L199

The input here is
```
[np.array([[10, 10], [12, 11]]), np.array([[10, 10], [13, 11]])]
```

**Comment by grfrederic** [comment]
Yes, I've had a bit more time to play with it. I have tried running the automatic test on my own too, and it worked. 

I have attached an example dataframe (CSV file) where the issue occurs, and a plot of my result. 

Here is the code I used to create the tracks:

```
data = pd.read_csv('~/tmp/example.csv')

max_distance = 15

lt = LapTrack(
    track_cost_cutoff=15**2,
    splitting_cost_cutoff=15**2,
    merging_cost_cutoff=15**2,
)

track_df, split_df, merge_df = lt.predict_dataframe(
    data,
    coordinate_cols=["h"], 
    frame_col="seconds",
    only_coordinate_cols=False,  
)
```

Finally, I noticed that setting some of the costs fixes the issue: for example, if I add `track_start_cost=1` i get two tracks just as I would expect.

![example](https://github.com/yfukai/laptrack/assets/26434160/b18c4278-b81f-437b-a0d0-b8b833225d7c)
[example.csv](https://github.com/yfukai/laptrack/files/12606352/example.csv)

**Comment by yfukai** [comment]
Thanks @grfrederic ! I'll definitely check it but I'm in a vacation and it might take a while.

**Comment by grfrederic** [comment]
No problem, you enjoy your vacation :) I'll use the small noise workaround for now, it works well. Should we reopen this issue?

**Comment by yfukai** [comment]
Yeah let's reopen this!

**Comment by yfukai** [comment]
@grfrederic I found the cause. It was a bug only happens when there's only a single non-moving particle in the coordinate. Will be fixed in #372. Thanks for your report again!

**Comment by grfrederic** [comment]
awesome, thank you for your help!
